part-2.py

```diff
@@ -4,13 +4,37 @@
 # the appropriate comment.
 
 # search
-
-
+def search(array, query):
+    if len(array) == 0: 
+        return False
+    else: 
+        if query == array[0]:
+            return True
+        else: 
+            return search(array[1:], query)
 
 # is_palindrome
-
-
+def is_palindrome(text): 
+    if not text: 
+        return False
+    if len(text) == 1: 
+        return True
+    else: 
+        if text[0] == text[-1]: 
+            return is_palindrome(text[1:-1])
 
 # digit_match
-
+def digit_match(a, b): 
```

Comment:
Nice solution using % and // The other way to do it is to convert the # to a string and take slices.